brave/brave-core
# Redesign Brave Wallet main layout introducing latest material 2 components

## Resolves https://github.com/brave/brave-browser/issues/32197


https://github.com/brave/brave-core/assets/3308503/759ac54a-dfa6-4606-a188-b40fe4c8289d



* Remove logic for old Android versions below Nougat (< API level 24).
* Switch from old `ViewPager` to new and more performant `ViewPager2` class.
* Switch from deprecated `FragmentStatePagerAdapter` to new `FragmentStateAdapter`.

* Rename `Market` section to `Explore` section per Figma design.
* Implement `BottomNavigationViewIndicator` to display sections: Portfolio, NFTs, Activity, Accounts, Explore.
**⚠️ Note:** NFTs section will be removed from the bottom navigation view and it will be merged into the portfolio section in a follow up PR (Main umbrella Issue: https://github.com/brave/brave-browser/issues/32196)

<img width="713" alt="Screenshot_2023-08-11_at_2_12_30_PM" src="https://github.com/brave/brave-core/assets/3308503/06d0c4e1-6ede-4a31-959c-82ec79e0a7ab">



## UI Changes

### Bottom Navigation View

Remove tab layout in favor of [bottom navigation view](https://m2.material.io/components/bottom-navigation) with gradient indicator that animates when tapped.

https://github.com/brave/brave-core/assets/3308503/5751b96a-d5ee-42ee-a3a8-02a83d83639d


### App Bar Layout

Add [app bar layout](https://m2.material.io/components/app-bars-top) and rewrite layout to scroll correctly.


https://github.com/brave/brave-core/assets/3308503/e0bb2e6e-47ce-4836-b252-762a0241ca1e


### Tweak FAB behavior

FAB "follows" bottom navigation view

https://github.com/brave/brave-core/assets/3308503/407c5a50-b502-4c3e-9b73-91bd6f974bb1

### Banner layout

Improved layout with bottom margin and thin separator.

<img width="713" alt="Screenshot_2023-08-11_at_2_27_07_PM" src="https://github.com/brave/brave-core/assets/3308503/50da8e85-34c3-455d-9301-2e5cac7c82d8">





## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

* Open Brave Wallet.
* Observe the new bottom navigation view is displayed instead of the old tabs on top.
* Tap on its sections.
* Observe the pages switch accordingly.
* Navigate back to Portfolio section and scroll down
* Observe the bottom navigation view disappears.
* Observe the FAB button animation
* Observe the toolbar can disappear
* Scroll up
* Observe the bottom navigation view is shown again
* Observe the FAB "follows" the bottom navigation view
* Observe the toolbar reappears.
